sqlstore/trades.go

```diff
@@ -110,15 +110,12 @@ func (ts *Trades) GetByMarket(ctx context.Context, market string, p entities.Off
 	return trades, nil
 }
 
-func (ts *Trades) GetByMarketWithCursor(ctx context.Context, market string, pagination entities.CursorPagination) ([]entities.Trade, entities.PageInfo, error) {
+func (ts *Trades) GetByMarketWithCursor(ctx context.Context, market string, pagination entities.CursorPagination) entities.ConnectionData[*v2.TradeEdge, entities.Trade] {
 	query := `SELECT * from trades WHERE market_id=$1`
-	args := []interface{}{entities.NewMarketID(market)}
-	trades, pageInfo, err := ts.queryTradesWithCursorPagination(ctx, query, args, pagination)
-	if err != nil {
-		return nil, pageInfo, fmt.Errorf("failed to get trade by market:%w", err)
-	}
+	countQuery := `SELECT count(*) from trades WHERE market_id=$1`
```

Comment:
How will this perform when there is a lot of data in the table?    as its not time bound (as the select query with pagination is) I think it may need to load all chunks of the hypertable to calculate this which would be very slow, e.g. (a strangley approprated example): https://stackoverflow.com/questions/67953098/timescaledb-select-count-slow-on-hypertable

Reply:
yeah, having to filter by market as well means that function won't work.  I suspect there is not a solution beyond not having a count in the case of hypertable data, but worth looking around to check.

Reply:
I can't really find anything that helps or suggests a better way of doing the count. We could add start/end time constraint where end time can be optional and help timescale to restrict the partitions it looks in to retrieve the data, but we would need to update the GraphQL query too and ensure that at least the start time is provided, and possibly even restrict that so that any date older than X days/weeks/month etc., would result in an date out-of-range error.

For older data, we could introduce an historic data API which may take longer to run but not have the start date limitation?

I think if we want to do that, we should raise it as a separate ticket and make the appropriate changes to all the paged queries and introduce the appropriate historic data queries where necessary.